File: Scripts/results/generated_code_results/Flutter/Utility&PracticalTools/PackageDelivery/SendParcel_Delivery_methodScreen.dart
import 'package:flutter/material.dart';
class SendParcel_Delivery_methodScreen extends StatefulWidget {
@override
_SendParcel_Delivery_methodScreenState createState() => _SendParcel_Delivery_methodScreenState();
}
class _SendParcel_Delivery_methodScreenState extends State<SendParcel_Delivery_methodScreen> {
String selectedDeliveryMethod = '';
TextEditingController nameController = TextEditingController();
TextEditingController emailController = TextEditingController();
TextEditingController phoneController = TextEditingController();
TextEditingController addressController = TextEditingController();
@override
Widget build(BuildContext context) {
return Scaffold(
appBar: AppBar(
title: Text('Send parcel'),
backgroundColor: Colors.white,
foregroundColor: Colors.black,
),
body: SingleChildScrollView(
child: Padding(
padding: const EdgeInsets.all(16.0),
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Text(
'Delivery method',
style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold),
),
SizedBox(height: 16),
_buildDeliveryMethodOption(
'From door to parcel center',
'1- 2 days',
'assets/images/template.png',
),
_buildDeliveryMethodOption(
'From door to door',
'2- 3 days',
'assets/images/template.png',
),
SizedBox(height: 24),
Text(
'Recipient info',
style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold),
),
SizedBox(height: 16),
_buildTextField('Name', nameController),
_buildTextField('Email', emailController),
_buildTextField('Phone number', phoneController),
_buildTextField('Address', addressController),
],
),
),
),
bottomNavigationBar: BottomNavigationBar(
currentIndex: 1,
items: [
BottomNavigationBarItem(
icon: Icon(Icons.mail_outline),
label: 'My parcels',
),
BottomNavigationBarItem(
icon: Icon(Icons.send),
label: 'Send parcel',
),
BottomNavigationBarItem(
icon: Icon(Icons.location_on_outlined),
label: 'Parcel center',
),
],
),
floatingActionButton: FloatingActionButton.extended(
onPressed: () {
if (selectedDeliveryMethod.isNotEmpty &&
nameController.text.isNotEmpty &&
emailController.text.isNotEmpty &&
phoneController.text.isNotEmpty &&
addressController.text.isNotEmpty) {
Navigator.pushNamed(context, 'SendParcel_Parcel_size');
} else {
ScaffoldMessenger.of(context).showSnackBar(
SnackBar(content: Text('Please fill all the fields')),
);
}
},
label: Text('Next'),
icon: Icon(Icons.arrow_forward),
backgroundColor: Colors.black,
),
);
}
Widget _buildDeliveryMethodOption(String title, String subtitle, String imagePath) {
return GestureDetector(
onTap: () {
setState(() {
selectedDeliveryMethod = title;
});
},
child: Container(
margin: EdgeInsets.only(bottom: 16),
decoration: BoxDecoration(
border: Border.all(
color: selectedDeliveryMethod == title ? Colors.black : Colors.grey,
width: 2,
),
borderRadius: BorderRadius.circular(8),
),
child: ListTile(
leading: Image.asset(imagePath, width: 40, height: 40),
title: Text(title),
subtitle: Text(subtitle),
trailing: selectedDeliveryMethod == title
? Icon(Icons.check_circle, color: Colors.black)
: null,
),
),
);
}
Widget _buildTextField(String label, TextEditingController controller) {
return Padding(
padding: const EdgeInsets.only(bottom: 16.0),
child: TextField(
controller: controller,
decoration: InputDecoration(
labelText: label,
border: OutlineInputBorder(),
),
),
);
}
}

File: Scripts/results/generated_code_results/RN_gpt/Utility&PracticalTools/PackageDelivery/SendParcel_Delivery_methodScreen.js
import React from 'react';
import { View, Text, TextInput, TouchableOpacity, StyleSheet, Image, ImageBackground } from 'react-native';
import { useNavigation } from '@react-navigation/native';

const SendParcel_Delivery_methodScreen = () => {
  const navigation = useNavigation();

  const goToParcelSize = () => {
    navigation.navigate('SendParcel_Parcel_size');
  };

  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <Text style={styles.time}>9:41</Text>
        <View style={styles.icons}>
          <ImageBackground source={require('../assets/snack-icon.png')} style={styles.icon} />
          <ImageBackground source={require('../assets/snack-icon.png')} style={styles.icon} />
        </View>
      </View>
      
      <Text style={styles.title}>Send parcel</Text>
      <Text style={styles.subtitle}>Delivery method</Text>

      <TouchableOpacity style={styles.card}>
        <Image source={require('../assets/snack-icon.png')} style={styles.cardIcon} />
        <View>
          <Text style={styles.cardTitle}>From door to parcel center</Text>
          <Text style={styles.cardDetails}>1 - 2 days</Text>
        </View>
      </TouchableOpacity>

      <TouchableOpacity style={styles.card} onPress={goToParcelSize}>
        <Image source={require('../assets/snack-icon.png')} style={styles.cardIcon} />
        <View>
          <Text style={styles.cardTitle}>From door to door</Text>
          <Text style={styles.cardDetails}>2 - 3 days</Text>
        </View>
      </TouchableOpacity>

      <Text style={styles.recipientInfo}>Recipient info</Text>

      <Text style={styles.label}>Name</Text>
      <TextInput style={styles.input} value="James May" />

      <Text style={styles.label}>Email</Text>
      <TextInput style={styles.input} value="james.rn" />

      <Text style={styles.label}>Phone number</Text>
      <TextInput style={styles.input} />

      <Text style={styles.label}>Address</Text>
      <TextInput style={styles.input} />

      <View style={styles.bottomNav}>
        <TouchableOpacity>
          <Image source={require('../assets/snack-icon.png')} style={styles.navIcon} />
          <Text style={styles.navLabel}>My parcels</Text>
        </TouchableOpacity>
        
        <TouchableOpacity>
          <Image source={require('../assets/snack-icon.png')} style={styles.navIcon} />
          <Text style={styles.navLabel}>Send parcel</Text>
        </TouchableOpacity>
        
        <TouchableOpacity>
          <Image source={require('../assets/snack-icon.png')} style={styles.navIcon} />
          <Text style={styles.navLabel}>Parcel center</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    padding: 20,
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  time: {
    fontSize: 16,
  },
  icons: {
    flexDirection: 'row',
  },
  icon: {
    width: 24,
    height: 24,
    marginLeft: 10,
  },
  title: {
    fontSize: 24,
    fontWeight: 'bold',
    marginTop: 30,
  },
  subtitle: {
    fontSize: 18,
    marginTop: 10,
  },
  card: {
    flexDirection: 'row',
    alignItems: 'center',
    padding: 15,
    borderWidth: 1,
    borderColor: '#ddd',
    borderRadius: 10,
    marginTop: 20,
  },
  cardIcon: {
    width: 40,
    height: 40,
    marginRight: 10,
  },
  cardTitle: {
    fontSize: 16,
  },
  cardDetails: {
    color: '#999',
  },
  recipientInfo: {
    fontSize: 18,
    marginTop: 40,
  },
  label: {
    fontSize: 14,
    marginTop: 10,
  },
  input: {
    borderWidth: 1,
    borderColor: '#ddd',
    borderRadius: 5,
    padding: 10,
    marginTop: 5,
  },
  bottomNav: {
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
    flexDirection: 'row',
    justifyContent: 'space-around',
    padding: 10,
    borderTopWidth: 1,
    borderColor: '#ddd',
  },
  navIcon: {
    width: 24,
    height: 24,
  },
  navLabel: {
    fontSize: 12,
    textAlign: 'center',
  },
});

export default SendParcel_Delivery_methodScreen;